calcom/cal.com
# fix: Detect correct timezone event on EventType component

## What does this PR do?
It detects the correct timezone from the client side in the preview of the Event Type scresidvermas/programming-knowledge?layout=mobile&month=2023-10&duration=15&date=2023-10-16en.


Fixes #11810 


![Screenshot from 2023-10-14 23-12-24](https://github.com/calcom/cal.com/assets/45305025/6a1e1198-de86-4f57-a9db-f0863ec27fd6)

## Requirement/Documentation



- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change




## How should this be tested?
1. Sign in
2. Select Event Types
3. Click on the preview icon
4. The timezone will be auto detected in the timezone picker

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self
